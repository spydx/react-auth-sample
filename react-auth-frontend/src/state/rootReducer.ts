import { combineReducers } from "redux";
import authReducer from './reducers/Auth';

export const rootReducer = combineReducers({
   auth: authReducer,
});



export type RootState = ReturnType<typeof rootReducer>;

