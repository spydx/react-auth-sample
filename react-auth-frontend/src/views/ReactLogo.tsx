
import logo from '../logo.svg';
import { 
   Container,
   Paper,
   Button,
   makeStyles,
   Typography
 } from '@material-ui/core';
import { useDispatch } from 'react-redux';
import { logout } from '../state/reducers/Auth';
import { useForm} from 'react-hook-form';
import '../App.css';

const useStyles = makeStyles((theme) => ({
   paper: {
     marginTop: theme.spacing(8),
     marginLeft: theme.spacing(1),
     marginRight: theme.spacing(1),
     display: 'flex',
     flexDirection: 'column',
     alignItems: 'center'
   },
   avatar: {
     margin: theme.spacing(1),
     backgroundColor: theme.palette.secondary.main,
   },
   form: { 
     width: '100%',
     marginTop: theme.spacing(1)
   },
   submit: {
     margin: theme.spacing(3,0,2),
   }
 
}));

export const ReactLogo = () => {
   const {handleSubmit} = useForm();
   const classes = useStyles();

   const dispatch = useDispatch();
  
   const onSubmit = () => {
      dispatch(logout(true));
   }

   return (
     <Container component="main" maxWidth="xs">
       <Paper className={classes.paper}>
     <div className={classes.paper}>
         <Typography component="h1" variant="h5">
           Logged in
        </Typography>
     <header className="App-header">
       <img src={logo} className="App-logo" alt="logo" />
       <p>
        
       </p>
       <form className={classes.form} onSubmit={handleSubmit(onSubmit)}>
          <Button 
               type="submit"
               fullWidth
               variant="contained"
               color="primary"
               className={classes.submit}
          >
            Logout
          </Button>
       </form>
       
     </header>
   </div>
     </Paper>
   </Container>
   );
 }