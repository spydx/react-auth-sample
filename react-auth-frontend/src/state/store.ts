import { CombinedState, configureStore } from "@reduxjs/toolkit";
import { rootReducer} from './rootReducer';
import { getDefaultMiddleware } from "@reduxjs/toolkit";
import logger from 'redux-logger';
import {AuthState } from '../state/reducers/Auth';

const customMiddelware = getDefaultMiddleware();
const middelware = [...customMiddelware, logger]

export const store = configureStore({
   reducer: rootReducer,
   middleware: middelware
});

const saveState = (state: CombinedState<AuthState>) => {

   try {
      if(state.token === null) {
         localStorage.removeItem('authtoken')
      } else {
         localStorage.setItem('authtoken', state.token)
      }
         
   } catch (error) {
      
   }
}
store.subscribe( () => {
   saveState(store.getState().auth);
})

export type AppDispatch = typeof store.dispatch;
