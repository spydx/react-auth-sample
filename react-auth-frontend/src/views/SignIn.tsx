import { 
   Container,
   Avatar,
   Button,
   TextField,
   Paper,
   makeStyles,
   CssBaseline,
   Typography,
   Box
} from '@material-ui/core';
import { LockOutlined } from '@material-ui/icons';
import { useForm} from 'react-hook-form';
import { useDispatch, useSelector } from 'react-redux';
import { loginSuccess, loginError } from '../state/reducers/Auth';
import {Â registerAccount, showRegisterView} from '../state/reducers/Account';
import { RootState } from '../state/rootReducer';

type LoginProps = {
   username: string
   password: string
}

export type LoggedIn = {
   token: string,
   tokenType: string, 
   profile: string,
}

export type RegisterAccount = {
   name: string,
   email: string,
   password: string,
}

const useStyles = makeStyles((theme) => ({
   paper: {
     marginTop: theme.spacing(8),
     marginLeft: theme.spacing(1),
     marginRight: theme.spacing(1),
     display: 'flex',
     flexDirection: 'column',
     alignItems: 'center'
   },
   avatar: {
     margin: theme.spacing(1),
     backgroundColor: theme.palette.secondary.main,
   },
   form: { 
     width: '100%',
     marginTop: theme.spacing(1)
   },
   submit: {
      marginTop: theme.spacing(1),
      marginBottom: theme.spacing(1)
   }
 
 }));


export const SignIn = () => {
   const classes = useStyles();
   const { register, handleSubmit } = useForm<LoginProps>();
   const dispatch = useDispatch();
   const account = useSelector((state: RootState) => state.account)
   const auth = useSelector((state: RootState) => state.auth)
   
   const onSubmit = async (login: LoginProps) => {
      const response = await fetch('http://localhost:8080/api/auth/login', {
        method: 'POST',
        headers: {'content-type': 'application/json'},
        body: JSON.stringify(login)
      })
      if(response.status !== 200) {
         console.log("triggede");
         dispatch(loginError(true));
         return
      }
      const data = await response.json()
      
      
      if(data?.token) {
         dispatch(loginSuccess(data.token));
      } 
      
      
   }

   const onRegister = async(reg: RegisterAccount ) => {
      const response = await fetch('http://localhost:8080/api/auth/register', {
         method: 'POST',
         headers: { 'Content-type': 'application/json'},
         body: JSON.stringify(reg)
      })
      const data = await response.json()
      if(data?.email && data?.name) { 
         dispatch(registerAccount(reg))
      }
   }

   const showRegistering = () => {
      dispatch(showRegisterView(true))
   }


   return(
      <Container component="main" maxWidth="xs">
         
         <CssBaseline />
         <Paper className={classes.paper}>
         <div className={classes.paper}>
            <Avatar className={classes.avatar}>
               <LockOutlined />
            </Avatar>
            <Typography component="h1" variant="h5">
              { account.registered ?  "Sign In" : "Register" }
            </Typography>
            <form className={classes.form} noValidate onSubmit={account.registered ? handleSubmit(onSubmit) : handleSubmit(onRegister)}>
               {account.registered ? null : <TextField 
                  margin="normal"
                  variant="outlined"
                  required
                  fullWidth
                  id="name"
                  name="name"
                  label="Name"
                  inputRef={register({required: "Must have name field"})}
                  /> }
               <TextField 
                  margin="normal"
                  variant="outlined"
                  required
                  fullWidth
                  id="email"
                  name="email"
                  label="Email address"
                  autoComplete="email"
                  autoFocus
                  inputRef={register({required: "Must have email field"})}
                  />
               <TextField 
                  variant="outlined"
                  fullWidth
                  required
                  id="password" 
                  type="password"
                  name="password"
                  label="Password"
                  inputRef={register({required: "Must have password field"})}
                  />
               { account.registered ? <Button 
                  type="submit"
                  fullWidth
                  variant="contained"
                  color="primary"
                  className={classes.submit}
               >Sign in</Button> : null }
               { account.registered ? <Button 
                  onClick={() => showRegistering()}
                  fullWidth
                  variant="contained"
                  color="primary"
                  className={classes.submit}
                  > Register </Button> : 
                  <Button 
                  type="submit"
                  fullWidth
                  variant="contained"
                  color="primary"
                  className={classes.submit}
                  > Register </Button>
               }
            </form>
         </div>
         { auth.error ? <Box bgcolor="error.main" p={2}><Typography component="h1" variant="h5" >
              Something went wrong, try again
         </Typography></Box>: null }
         </Paper>
      </Container>
   );
}

