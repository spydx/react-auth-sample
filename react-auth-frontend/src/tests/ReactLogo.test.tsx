
import { ReactLogo } from '../views/ReactLogo';
import { store } from '../state/store';
import { Provider } from 'react-redux';
import { create} from 'react-test-renderer';
import { render } from '@testing-library/react';

const component = <Provider store={store}><ReactLogo /></Provider>

describe('A snapshot test of ReactLogo', () => {
  test('Render ReactLogo', () => {
    let tree = create(component)
    expect(tree.toJSON()).toMatchSnapshot();
  })

  test('Render Logout', () => {
   const { getByText } = render(component)
   const logout  = getByText("Logout")
   expect(logout).toBeInTheDocument();

 })
})