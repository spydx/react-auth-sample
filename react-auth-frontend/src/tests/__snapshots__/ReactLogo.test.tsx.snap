// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A snapshot test of ReactLogo Render ReactLogo 1`] = `
<main
  className="MuiContainer-root MuiContainer-maxWidthXs"
>
  <div
    className="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      className="makeStyles-paper-1"
    >
      <h1
        className="MuiTypography-root MuiTypography-h5"
      >
        Logged in
      </h1>
      <header
        className="App-header"
      >
        <img
          alt="logo"
          className="App-logo"
          src="logo.svg"
        />
        <p />
        <form
          className="makeStyles-form-3"
          onSubmit={[Function]}
        >
          <button
            className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-submit-4 MuiButton-containedPrimary MuiButton-fullWidth"
            disabled={false}
            onBlur={[Function]}
            onDragLeave={[Function]}
            onFocus={[Function]}
            onKeyDown={[Function]}
            onKeyUp={[Function]}
            onMouseDown={[Function]}
            onMouseLeave={[Function]}
            onMouseUp={[Function]}
            onTouchEnd={[Function]}
            onTouchMove={[Function]}
            onTouchStart={[Function]}
            tabIndex={0}
            type="submit"
          >
            <span
              className="MuiButton-label"
            >
              Logout
            </span>
          </button>
        </form>
      </header>
    </div>
  </div>
</main>
`;
